tls/s2n_security_rules.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include "stuffer/s2n_stuffer.h"
+#include "utils/s2n_result.h"
+
+typedef enum {
+    S2N_PERFECT_FORWARD_SECRECY = 0,
+    S2N_SECURITY_RULES_COUNT,
+} s2n_security_rule_id;
+
+#define TO_FLAG(rule) rule##_FLAG = (1 << (rule))
+typedef enum {
+    TO_FLAG(S2N_PERFECT_FORWARD_SECRECY),
+} s2n_security_flag;
+
+struct s2n_security_rule_result {
+    bool found_error;
```

Comment:
I would prefer to invert this logic and have something like `bool is_valid`, `bool passed` or `bool success`.

Reply:
The trick there is that we then need to initialize is_valid to "true" before calling any of our validation methods. It'll default to incorrectly start as "false". If we use the negative case, it starts out correct and methods just need to set it if they find an issue. Like, our checks don't ever definitively mark a policy as "valid", they only ever flag it as "not valid", and if they never flag it then it's valid.